File: Algorithms/Python/Unique-Binary-Search-Trees.py
class Solution(object):

    def numTrees(self, n):
        """
        :type n: int
        :rtype: int
        """
        ans = [1, 1]
        for i in range(2, n + 1):
            temp = 0
            for j in range(1, i + 1):
                temp = temp + ans[j - 1] * ans[i - j]
            ans.append(temp)
        return ans[n]

File: Algorithms/Java/Unique Binary Search Trees.java
public class Solution {
    public int numTrees(int n) {
        int[] ans = new int[n + 1];
		ans[0] = ans[1] = 1;
		for (int i = 2; i < n + 1; i++) {
			for (int j = 1; j < i + 1; j++)
				ans[i] += (ans[j - 1] * ans[i - j]);
		}
		return ans[n];

    }
}